### PR TITLE
refactor: replace manual JSON serialization with JObject-based approach

### PR DESCRIPTION
**Pull request by levtoji**
## Summary

Replaced the hand-rolled StringBuilder-based JSON serialization in `EdiObject` with JObject/JArray construction + `JsonConvert.SerializeObject()`. No more manual JSON building.

### `IEdiObject.cs` (-199 / +70 lines)

**Removed:**
- `escapeSpecialChars()` -- custom JSON string escaping is now handled by JsonConvert
- `_builder` field -- no more shared StringBuilder state
- `RecurseJSON(EdiObject)` -- replaced by `BuildJToken()` + `AddChildrenToJObject()`
- `RecurseJSON(XElement)` -- dead code, no callers existed
- `_builder.Replace()` double-escape workaround

**Added:**
- `BuildJToken(EdiObject)` -- recursive JObject/JArray tree builder, handles multi-value field `JArray` expansion
- `AddChildrenToJObject(JObject, List<EdiObject>)` -- child grouping by name into JArrays

### `SerializeToJsonTests.cs` (+127 lines)

**5 structural assertion tests** -- verify specific properties/keys in the JSON output:
- Simple fields + key
- Children grouped by name  
- Empty object edge case (Kontakt: no fields, no children)
- Multi-value field expansion
- Multi-value with children attached to last object

**5 snapshot comparison tests** -- compare new output against JSON snapshots generated from the old master code via `JToken.DeepEquals()` (structural/indentation-agnostic):
- Snapshot files stored in `EDILibraryTests/Snapshots/*.json`
- Generated from old code to guarantee semantic equivalence

## Verification

All 130 tests pass (7 existing + 10 new).

### CODE REVIEW

**Comment by levtoji** [comment]
Antwort auf https://github.com/Hochfrequenz/EDILibrary/commit/5f7052eab37c52b31d494e2b685fa14ef0a1756e

**Comment by hf-kklein** [comment]
@levtoji warum gehen wir noch auf Newtonsoft Json und nicht direkt System.Text.json? DAs ist ja die Zukunft.

**Comment by levtoji** [comment]
Weils ein viel größeres Refactoring benötigt hätte, kann man aber auch mal probieren, wollte erstmal mit Newtonsoft schauen was geht.

File: EDILibraryTests/SerializeToJsonTests.cs [hunk 1]
```diff
@@ -102,5 +104,173 @@ public void SerializeToJSON_WithMultipleControlCharsInField_ProducesValidJson()
...
+            obj.Fields["Name"] = new List<string> { "Wert" };
+            obj.Fields["Zaehler"] = new List<string> { "42" };
+
+            var json = obj.SerializeToJSON();
+
+            using var doc = JsonDocument.Parse(json);
+            var elem = doc.RootElement.GetProperty("Dokument")[0];
+            Assert.AreEqual("abc", elem.GetProperty("Key").GetString());
+            Assert.AreEqual("Wert", elem.GetProperty("Name").GetString());
+            Assert.AreEqual("42", elem.GetProperty("Zaehler").GetString());
```
Comment (hf-kklein) [inline EDILibraryTests/SerializeToJsonTests.cs]: kannst du hier awesome assertions verwenden? dann liest es sich schöner :)

Comment (levtoji) [reply EDILibraryTests/SerializeToJsonTests.cs]: @copilot Bitte umsetzen in awesome asseritions

File: EDILibraryTests/SerializeToJsonTests.cs [hunk 2]
```diff
@@ -102,5 +105,173 @@ public void SerializeToJSON_WithMultipleControlCharsInField_ProducesValidJson()
...
+            obj.AddChild(child1);
+            obj.AddChild(child2);
+
+            var json = obj.SerializeToJSON();
+
+            using var doc = JsonDocument.Parse(json);
+            var segments = doc.RootElement.GetProperty("Dokument")[0].GetProperty("Segment");
+            Assert.AreEqual(2, segments.GetArrayLength());
+            Assert.AreEqual("A", segments[0].GetProperty("Code").GetString());
+            Assert.AreEqual("B", segments[1].GetProperty("Code").GetString());
```
Comment (hf-kklein) [inline EDILibraryTests/SerializeToJsonTests.cs]: hier sind doch immer noch nicht-awesome assertions...

Comment (levtoji) [reply EDILibraryTests/SerializeToJsonTests.cs]: lol

File: EDILibraryTests/SerializeToJsonTests.cs [hunk 3]
```diff
@@ -100,7 +103,177 @@ public void SerializeToJSON_WithMultipleControlCharsInField_ProducesValidJson()
...
+                .BeTrue(
+                    $"New output for '{snapshotName}' differs from snapshot.\nSnapshot: {snapshot}\nNew:      {newOutput}"
+                );
+        }
+
+        private static JToken LoadSnapshot(string name)
+        {
+            var path = Path.GetFullPath(Path.Combine("../../../Snapshots", name + ".json"));
+            var json = File.ReadAllText(path);
+            return JToken.Parse(json);
```
[the remark this answers is not in the document]
Comment (levtoji) [reply EDILibraryTests/SerializeToJsonTests.cs:227]: bitte implementieren @copilot

**Review by hf-kklein — APPROVED** [review]
lgtm. jetzt muss man halt den snapshot ggf. händisch aktualisieren. dachte ursprünglich, wir könnten ein framework wie verify oder so verwenden aber ok - so häufig wird sich die json serialisierung ja nicht ändern in der lebenszeit dieser library. hast du die snapshots erzeugt _BEVOR_ du auf Newtonsoft umgestellt hast? Ich nehme an ja, sonst wär's ja witzlos


bleibt noch der umstieg auf System.Text.Json als nächstes todo

**Comment by levtoji** [comment]
> hast du die snapshots erzeugt BEVOR du auf Newtonsoft umgestellt hast? 

Ja genau die snapshots sind vor der code anpassung entstanden.
